miso-web/src/main/webapp/scripts/workflow_display.js

```diff
@@ -84,29 +85,54 @@ var WorkflowDisplay = (function() {
   }
 
+  function makeLogIconCell(file, label) {
+    var cell = jQuery("<td class='logIconCell'" + (label ? " title='" + label + "'" : "") + ">");
+    if (file) {
+      cell.append(jQuery("<img src='/styles/images/" + file + "' class='logIcon'>"));
+    } else {
+      cell.append(jQuery("<div class='logIcon'>"));
+    }
+    return cell;
+  }
+
   function makeArrowCell(entryStepNumber, currentStepNumber) {
-    return entryStepNumber === currentStepNumber ? jQuery("<td>").append(jQuery("<img src='/styles/images/arrow.svg' class='logIcon'>"))
-        : jQuery("<td>");
+    return entryStepNumber === currentStepNumber ? makeLogIconCell('arrow-right.svg') : makeLogIconCell();
   }
 
-  function makeModifyCell(entryStepNumber, numEntries) {
-    return entryStepNumber === numEntries - 1 ? jQuery("<td>").append(jQuery("<img src='/styles/images/cancel.png' class='logIcon'>"))
-        : jQuery("<td>").append(jQuery("<img src='/styles/images/redo.svg' class='logIcon'>"));
+  function makeActionCell(entryStepNumber, numEntries, workflowId) {
+    if (entryStepNumber === numEntries - 1) {
+      return makeLogIconCell('arrow-undo.svg', 'Undo').click(function() {
+        cancelInput(workflowId);
+      });
+    } else {
+      return makeLogIconCell('arrow-back.svg', 'Go Back').click(function() {
+        changeStep(workflowId, entryStepNumber);
+      });
+    }
   }
 
-  function makeLogEntryClickHandler(entryStepNumber, numEntries, workflowId) {
-    return function() {
-      entryStepNumber === numEntries - 1 ? cancelInput(workflowId) : changeStep(workflowId, entryStepNumber)
-    };
+  function makeRow(rowNum, cells) {
+    return jQuery("<tr class=" + (rowNum % 2 === 1 ? "odd" : "even") + ">").append(cells);
   }
 
-  function makeLog(logEntries, workflowId, currentStepNumber) {
+  function makeLog(logEntries, workflowId, currentStepNumber, complete) {
     var table = jQuery("<table class='workflowLogTable'>");
+    var rowNum = 1;
+
+    if (currentStepNumber != null && currentStepNumber < logEntries.length) {
+      var arrowCell = makeArrowCell(-1, currentStepNumber);
+      var messageCell = jQuery("<td>").text(complete ? "Complete workflow" : "Resume workflow");
+      var actionCell = makeLogIconCell('arrow-play.svg', 'Return').click(function() {
+        continueWorkflow(workflowId);
+      });
+      table.append(makeRow(rowNum++, [arrowCell, messageCell, actionCell]));
+    }
 
-    for (var i = 0; i < logEntries.length; i++) {
-      table.prepend(jQuery("<tr>").append(
-          [makeArrowCell(i, currentStepNumber), jQuery("<td>").text(logEntries[i]), jQuery("<td>"), makeModifyCell(i, logEntries.length)])
-          .click(makeLogEntryClickHandler(i, logEntries.length, workflowId)));
+    for (var i = logEntries.length - 1; i >= 0; i--) {
+      var arrowCell = makeArrowCell(i, currentStepNumber);
+      var messageCell = jQuery("<td>").text(logEntries[i]);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "messageCell" as this name is already used in declaration at line 124. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)